### DISCUSSION TITLE
Doc comment on text2vec-huggingface (/modules/retriever-vectorizer-modules/text2vec-huggingface)

### DISCUSSION THREAD

**Discussion by giscus** [weaviate/weaviate-io#712] [Docs/blog comments]
# Doc comment on text2vec-huggingface (/modules/retriever-vectorizer-modules/text2vec-huggingface)

Introduction

https://weaviate.io/developers/weaviate/modules/retriever-vectorizer-modules/text2vec-huggingface

**Comment by carlesoctav** [comment]
"Does this modules support non-transformers models such as ONNX or Keras on hugging face?
or it's only for model that created from transformers API?

**Reply by prrao87** [reply]
If my understanding is correct, I believe you'd need to write your own `text2vec-onnx` model to accomplish this. You'd then attach the custom module to a FastAPI `upsert` endpoint (or similar) and then when you call that endpoint, the custom `text2vec` module would be used to index data to Weaviate.

Example of custom `text2vec-spacy` module and FastAPI setup here:
https://github.com/laura-ham/t2v-spacy-models

I'm planning on using this template to build an ONNX vectorization pipeline to Weaviate myself 😅
